src/webclients/components/sdfv/properties/properties.ts

```diff
@@ -171,27 +187,28 @@ export class KeyProperty {
 export class ValueProperty extends Property {
 
     constructor(
-        element: any | undefined,
-        xform: any | undefined,
-        target: any,
+        element: WithAttributes | undefined,
+        xform: JsonTransformation | undefined,
+        target: Record<string, any>,
         key: string,
         subkey: string | undefined,
         datatype: string,
-        protected input: JQuery<HTMLElement>
+        protected input: JQuery
     ) {
         super(element, xform, target, key, subkey, datatype);
     }
 
     public getValue(): PropertyValueReturn {
-        let value: any = this.input.is(':checkbox') ?
+        let value: unknown = this.input.is(':checkbox') ?
             this.input.is(':checked') : this.input.val();
 
         if (this.datatype === 'LambdaProperty') {
             if (value === '' || value === undefined)
                 value = null;
         } else if (this.input.attr('type') === 'number') {
             try {
-                value = parseInt(value);
+                if (typeof value === 'string')
+                    value = parseInt(value);
```

Comment:
this can create a situation in which value is not a `number` but still returned? you can keep this but I am not sure what is the purpose of this function.